_inc/client/appearance/index.jsx

```diff
@@ -37,16 +40,51 @@ export const Page = ( props ) => {
 		getModule
 	} = props,
 		isAdmin = props.userCanManageModules,
+		infiniteScrollDesc = getModule( 'infinite-scroll' ).description,
+		noInfiniteScrollSupport = ! props.themeData.support.infiniteScroll,
 		moduleList = Object.keys( props.moduleList );
 
+	if ( noInfiniteScrollSupport ) {
+		infiniteScrollDesc =
+			<div>
+				{ infiniteScrollDesc }
+				<p className="jp-form-setting-explanation">
+					{
+						__( "Your theme %(theme)s doesn't currently support Infinite Scroll, which needs data from your theme to function properly.",
+							{
+								args: {
+									theme: props.themeData.name
+								}
+							}
+						)
+					}
+				</p>
+				{
+					props.themeData.hasUpdate
+						? <p className="jp-form-setting-explanation">
```

Comment:
When I need to return a component within a ternary, I usually try and format it like this:

```
? (
	<p className="jp-form-setting-explanation">
	{
		__( "There's an update available for your theme. Check if this update adds Infinite Scroll support in {{a}}WordPress Updates{{/a}}.",
			{
				components: {
					a: <a href={ props.siteAdminUrl + 'update-core.php' } className="jetpack-js-stop-propagation" />
				}
			}
		)
	}
	</p>
)
```

Not a blocker. Just making a note.